.meta/_partials/fields/_compression_options.toml.erb

```diff
@@ -3,17 +3,32 @@
     "common" => true,
     "default" => "none",
     "description" => """\
-The compression strategy used to compress the encoded event data before \
+The compression algorithm used to compress the encoded event data before \
 transmission.\
 """
 }.merge(options || {}) -%>
+
 <%- enum = {
     "none" => "No compression.",
     "gzip" => "[Gzip][urls.gzip] standard DEFLATE compression."
 }.merge(enum || {}) -%>
 
 [<%= namespace %>.compression]
+type = "table"
+description = """\
+Configures the compression specific sink behavior.\
+"""
+
+[<%= namespace %>.compression.children.codec]
 <%= options.to_toml %>
 
-[<%= namespace %>.compression.enum]
+[<%= namespace %>.compression.children.codec.enum]
 <%= enum.to_toml %>
+
+[<%= namespace %>.compression.children.level]
+type = "uint"
+default = 6
+examples = [0, 1, 6, 9]
```

Comment:
Agree. We can just remove the default and note each algorithm's default in the description. Also, the `type` should be changed to `any` since it allows for integers and strings.

Reply:
It'd be nice if we could present unions in the docs so that the type could be something like `string | uint8`.